ironfish/src/account/accounts.ts

```diff
@@ -603,6 +598,12 @@ export class Accounts {
 
     const accountHeadHash = this.chainProcessor.hash
 
+    if (scan.isAborted) {
+      scan.signalComplete()
+      this.scan = null
+      return
+    }
```

Comment:
Once we're done waiting on the updateHead() lock, we need to check if we should abort, instead of start iterating the chain. This could happen during a node shutdown.